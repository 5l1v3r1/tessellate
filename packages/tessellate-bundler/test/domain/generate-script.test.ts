import fs = require('mz/fs');
import path = require('path');
import generateScript from '../../src/domain/generate-script';

describe('script-builder', () => {
  (jest as any).setTimeout(30000);

  it('should build a fragment script string from a content structure', async () => {
    const json = await fs.readFile(path.resolve(__dirname, '../fixtures/content.json'));
    const layout = JSON.parse(json.toString());
    const result = generateScript({ layout, content: {} });
    expect(typeof result).toBe('string');
    expect(result).toMatchSnapshot();
  });

  it('should build a fragment script string with multiple component imports', async () => {
    const layout = {
      type: 'node-module-a.MyComponent1',
      props: null,
      children: [
        {
          type: 'node-module-a.MyComponent2',
          props: null,
          children: []
        },
        {
          type: 'node-module-a.MyComponent2',
          props: null,
          children: []
        },
        {
          type: 'node-module-b.MyComponent.FooBar',
          props: null,
          children: []
        }
      ]
    };
    const result = generateScript({ layout, content: {} });
    expect(typeof result).toBe('string');
    expect(result).toMatchSnapshot();
  });
});
