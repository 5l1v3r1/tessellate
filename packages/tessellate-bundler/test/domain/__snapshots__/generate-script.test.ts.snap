// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`script-builder should build a fragment script string from a content structure 1`] = `
"import React from \\"react\\";
import ReactDOM from \\"react-dom\\";
import mosaicComponentExample from \\"mosaic-component-example\\";

export const Root = props => {
  return (
    <div id=\\"root\\" data-props={JSON.stringify(props)}>
      {getElements(props)}
    </div>
  );
};

const getElements = props => {
  const bundledProps = {};
  const mergedProps = Object.assign({}, bundledProps, props);
  const getProps = pointerFn => pointerFn(mergedProps);
  return React.createElement(
    \\"div\\",
    null,

    React.createElement(mosaicComponentExample, { level: 3 }, \`Hello, World!\`)
  );
};

export default function render(element) {
  const props = JSON.parse(element.getAttribute(\\"data-props\\"));
  ReactDOM.render(getElements(props), element);
}
"
`;

exports[`script-builder should build a fragment script string with multiple component imports 1`] = `
"import React from \\"react\\";
import ReactDOM from \\"react-dom\\";
import { MyComponent1, MyComponent2 } from \\"node-module-a\\";
import { MyComponent } from \\"node-module-b\\";

export const Root = props => {
  return (
    <div id=\\"root\\" data-props={JSON.stringify(props)}>
      {getElements(props)}
    </div>
  );
};

const getElements = props => {
  const bundledProps = {};
  const mergedProps = Object.assign({}, bundledProps, props);
  const getProps = pointerFn => pointerFn(mergedProps);
  return React.createElement(
    MyComponent1,
    null,

    React.createElement(MyComponent2, null),
    React.createElement(MyComponent2, null),
    React.createElement(MyComponent, null)
  );
};

export default function render(element) {
  const props = JSON.parse(element.getAttribute(\\"data-props\\"));
  ReactDOM.render(getElements(props), element);
}
"
`;
